packages/google_maps_flutter/android/src/main/java/io/flutter/plugins/googlemaps/GoogleMapBuilder.java

```diff
@@ -14,11 +14,12 @@
 class GoogleMapBuilder implements GoogleMapOptionsSink {
   private final GoogleMapOptions options = new GoogleMapOptions();
   private boolean trackCameraPosition = false;
+  private boolean myLocationEnabled = false;
 
   GoogleMapController build(
       int id, Context context, AtomicInteger state, PluginRegistry.Registrar registrar) {
     final GoogleMapController controller =
-        new GoogleMapController(id, context, state, registrar, options);
+        new GoogleMapController(id, context, state, registrar, options, myLocationEnabled);
```

Comment:
Can you explain again why can't we follow the same pattern as setTrackCameraPosition...
e.g after `controller.init` call `controller.setMyLocationEnabled` ?

Reply:
Thanks for the explanation, now I get it 😄 

What do you think about calling an `updateMyLocationEnabled()` from `onMapReady` which will set the map's my location enabled field to whatever the value of the member `myLocationEnabled` is.
Then have `setMyLocationEnabled` to do nothing if the new value is the same as the current one, or if we're not yet initialized, otherwise update `myLocationEnabled` (the member) and call `updateMyLocationEnabled()`

Reply:
Okay I did something like that. `setMyLocationEnabled` skips `updateMyLocation` if the map is null, and during map initiation `updateMyLocation` is called.

 From the [docs](https://developers.google.com/android/reference/com/google/android/gms/maps/GoogleMap#setMyLocationEnabled(boolean)) its unclear to me if you need to check permissions to `setMyLocationEnabled` to `false`, so I check them no matter if you are turning it on or off, that might not be right.